src/auth/auth-config.ts

```diff
@@ -0,0 +1,576 @@
+  public readonly providerId: string;
+  public readonly idpEntityId: string;
+  public readonly ssoURL: string;
+  public readonly x509Certificates: string[];
+  public readonly rpEntityId: string;
+  public readonly callbackURL?: string;
+  public readonly enableRequestSigning?: boolean;
+
+  /**
+   * Static method convert a client side request to a SAMLConfigServerRequest.
+   * Throws an error if validation fails. If the request is not a SAMLConfig request,
+   * returns null.
+   *
+   * @param {SAMLAuthProviderRequest} options The options object to convert to a server request.
+   * @param {boolean=} ignoreMissingFields Whether to ignore missing fields.
+   * @return {?SAMLConfigServerRequest} The resulting server request or null if not valid.
+   */
+  public static buildServerRequest(
+      options: SAMLAuthProviderRequest,
+      ignoreMissingFields: boolean = false): SAMLConfigServerRequest | null {
+    const makeRequest = typeof options !== 'undefined' &&
+        validator.isNonNullObject(options) &&
+        (options.providerId || ignoreMissingFields);
+    if (!makeRequest) {
+      return null;
+    }
+    const request: SAMLConfigServerRequest = {};
+    // Validate options.
+    SAMLConfig.validate(options, ignoreMissingFields);
+    request.enabled = options.enabled;
+    request.displayName = options.displayName;
+    // IdP config.
+    if (options.idpEntityId || options.ssoURL || options.x509Certificates) {
+      request.idpConfig = {
+        idpEntityId: options.idpEntityId,
+        ssoUrl: options.ssoURL,
+        signRequest: options.enableRequestSigning,
+        idpCertificates: typeof options.x509Certificates === 'undefined' ? undefined : [],
+      };
+      if (options.x509Certificates) {
+        for (const cert of (options.x509Certificates || [])) {
+          request.idpConfig.idpCertificates.push({x509Certificate: cert});
+        }
+      }
+    }
+    // RP config.
+    if (options.callbackURL || options.rpEntityId) {
+      request.spConfig = {
+        spEntityId: options.rpEntityId,
+        callbackUri: options.callbackURL,
+      };
+    }
+    return request;
+  }
+
+  /**
+   * Returns the provider ID corresponding to the resource name if available.
+   *
+   * @param {string} resourceName The server side resource name.
+   * @return {?string} The provider ID corresponding to the resource, null otherwise.
+   */
+  public static getProviderIdFromResourceName(resourceName: string): string | null {
+    // name is of form projects/project1/inboundSamlConfigs/providerId1
+    const matchProviderRes = resourceName.match(/\/inboundSamlConfigs\/(saml\..*)$/);
+    if (!matchProviderRes || matchProviderRes.length < 2) {
+      return null;
+    }
+    return matchProviderRes[1];
+  }
+
+  /**
+   * @param {any} providerId The provider ID to check.
+   * @return {boolean} Whether the provider ID corresponds to a SAML provider.
+   */
+  public static isProviderId(providerId: any): boolean {
```

Comment:
Declare return type with a type guard on `providerId`.

`public static isProviderId(providerId: any): providerId is string {`

This way, once this method is called, `providerId` can be handled as a string at the callsite.